src/commands/reminders/remind.ts

```diff
@@ -11,38 +11,53 @@ export default class extends Command {
 	description = `Have ${BOT.NAME} give you a reminder.`;
 	usage = '<reminder> | <duration> | [repeat]';
 	extendedHelp = 'Reminders can be set to repeat daily or weekly.';
-
-	run(msg: Message, [reminder]: [Reminder]): Promise<Message> {
-		msg.client.mongo.collection(DB.REMINDERS).insertOne(reminder);
-
-		return msg.channel.send(`I'll remind you about that at ${reminderTime(reminder)}.`);
-	}
-
-	argParser(msg: Message, input: string): [Reminder] {
-		const [content, rawDuration, rawRepeat] = input.split('|').map(part => part.trim());
-		const weekWords = ['w', 'week', 'weekly'];
-		const dayWords = ['d', 'day', 'daily'];
-
+	options: ApplicationCommandOptionData[] = [
+		{
+			name: 'content',
+			description: 'What you\'d like to be reminded of',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'duration',
+			description: 'When you\'d like to be reminded',
+			type: 'STRING',
+			required: true
+		},
+		{
+			name: 'repeat',
+			description: 'How often you want the reminder to repeat',
+			choices: [{ name: 'daily', value: 'daily' }, { name: 'weekly', value: 'weekly' }],
+			type: 'STRING',
+			required: false
+		}
+	]
+
+	tempRun(interaction: CommandInteraction): Promise<unknown> {
+		const content = interaction.options.getString('content');
+		const rawDuration = interaction.options.getString('duration');
 		const duration = parse(rawDuration);
-		if (!duration) throw `**${rawDuration}** is not a valid duration.`;
```

Comment:
For this error I'd highly recommend specifying examples of valid durations for those who might not be familiar with Discord, let alone this command. I've done this [on this line of poll.ts](https://github.com/ud-cis-discord/SageV2/blob/efcb84a7264a2371a2ffc9a2d71b4fca172a1adc/src/commands/fun/poll.ts#L66); feel free to re-use that code here.